### DISCUSSION TITLE
Using GetText.NET with WPF / Xaml

### DISCUSSION THREAD

**Discussion by sdraht** [perpetualKid/GetText.NET#42] [Q&A] [answered]
Hello,
has anyone yet used GetText.NET for a WPF Application or other Xaml-Files?
It seems quite doable to extend GetText.NET in a clean way to also support WPF, just as GetText.NET WindowsForms does it for WinForms.
Actually, in contrast to the WindowsForms-approach, I would prefer to explicitly mark any translatable text in the Xaml, e.g. surround it by some MarkupExtension with additional context description, maybe some tricky mechanism to choose the right catalog per *.csproj.

Maybe somebody did this already in one of the various forks? Searching the internet only found solutions based on NGetText directly, but I really like the improvements added by this project.

**Comment by perpetualKid** [comment]
> Hello, has anyone yet used GetText.NET for a WPF Application or other Xaml-Files? It seems quite doable to extend GetText.NET in a clean way to also support WPF, just as GetText.NET WindowsForms does it for WinForms. 

I haven't heard of any such efforts, but given this project GetText.Net is based on NGetText and share most of the public API, it should be manageable attempt to port existing NGetText (or other libraries based on gettext-alike interfaces) WPF/Xaml helpers to GetText.Net

> ... maybe some tricky mechanism to choose the right catalog per *.csproj.

Not sure what is needed here, but there is also [CatalogManager](https://github.com/perpetualKid/GetText.NET/blob/main/src/GetText/CatalogManager.cs) which can handle multiple catalogs ie. per assembly (which should be equivalent to csproj).

**Comment by perpetualKid** [comment, marked as answer]
Handling WPF is now available in code, both for Extractor and also Wpf helper supporting runtime handling of translations in WPF application. 
Pending publishing to Nuget, will also be done shortly
